keyboards/handwired/dactyl_manuform/5x6_right_trackball/5x6_right_trackball.c

```diff
@@ -0,0 +1,202 @@
+/* Copyright 2020 Christopher Courtney, aka Drashna Jael're  (@drashna) <drashna@live.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "5x6_right_trackball.h"
+
+#ifndef TRACKBALL_DPI_OPTIONS
+#    define TRACKBALL_DPI_OPTIONS { 1200, 1600, 2400 }
+#    ifndef TRACKBALL_DPI_DEFAULT
+#        define TRACKBALL_DPI_DEFAULT 1
+#    endif
+#endif
+#ifndef TRACKBALL_DPI_DEFAULT
+#    define TRACKBALL_DPI_DEFAULT 0
+#endif
+
+keyboard_config_t keyboard_config;
+uint16_t dpi_array[] = TRACKBALL_DPI_OPTIONS;
+#define DPI_OPTION_SIZE (sizeof(dpi_array) / sizeof(uint16_t))
+
+bool     BurstState        = false;  // init burst state for Trackball module
+uint16_t MotionStart       = 0;      // Timer for accel, 0 is resting state
+
+__attribute__((weak)) void process_mouse_user(report_mouse_t* mouse_report, int16_t x, int16_t y) {
+    mouse_report->x = x;
+    mouse_report->y = y;
+}
+
+__attribute__((weak)) void process_mouse(report_mouse_t* mouse_report) {
+    report_pmw_t data = pmw_read_burst();
+    if (data.isOnSurface && data.isMotion) {
+        // Reset timer if stopped moving
+        if (!data.isMotion) {
+            if (MotionStart != 0) MotionStart = 0;
+            return;
+        }
+
+        // Set timer if new motion
+        if ((MotionStart == 0) && data.isMotion) {
+            if (debug_mouse) dprintf("Starting motion.\n");
+            MotionStart = timer_read();
+        }
+
+        if (debug_mouse) {
+            dprintf("Delt] d: %d t: %u\n", abs(data.dx) + abs(data.dy), MotionStart);
+        }
+        if (debug_mouse) {
+            dprintf("Pre ] X: %d, Y: %d\n", data.dx, data.dy);
+        }
```

Comment:
```suggestion
        #ifdef CONSOLE_ENABLE
        if (debug_mouse) {
            dprintf("Delt] d: %d t: %u\n", abs(data.dx) + abs(data.dy), MotionStart);
        }
        if (debug_mouse) {
            dprintf("Pre ] X: %d, Y: %d\n", data.dx, data.dy);
        }
        #endif
```
I'd probably do this with the other debug-related code as well.

Reply:
Shouldn't matter.  dprintf doesn't do anything if console isn't enabled, because debugging and debug printing is disabled.

Reply:
Specifically:
https://github.com/qmk/qmk_firmware/blob/54b932e84404725d3ac77d36faa2001c0aa8267b/tmk_core/common/debug.h#L145-L149

And https://github.com/qmk/qmk_firmware/blob/54b932e84404725d3ac77d36faa2001c0aa8267b/tmk_core/common.mk#L89-L94

Reply:
Which is to say, I can change this, I just don't feel it's needed.

Reply:
I know it will not do anything, that's why it doesn't make much sense to compile it.
Although the compiler is probably smart-enough to leave it out. (?)

In any case, I trust you on this, so I'm fine either way, hence the approval.

Reply:
Yeah, it should be smart enough to compile it out, as it doesn't do anything.

I just wanted to make sure I wasn't missed anything.